snap/snapcraft.yaml

```diff
@@ -359,6 +436,56 @@ parts:
     source: snap/local/build-helpers
     stage-packages:
       - curl
+
+  # snapcraft preload is necessary to make postgres just use a different
+  # lockfile location in /dev/shm
+  # NOTE: if this ever breaks in really scary ways when compiling with lots of
+  # warnings, see the comment on the kong part, tldr probably some other C/C++
+  # part in the build broke snapcraft-preload by running before
+  # snapcraft-preload
+  snapcraft-preload:
+    source: https://github.com/sergiusens/snapcraft-preload.git
```

Comment:
As there've been no releases of snapcraft-preload, should we pin this to a specific commit?  The README for this code doesn't fully explain how it works, especially with respect to the environment variable SNAPCRAFT_PRELOAD_REDIRECT_ONLY_SHM.

Reply:
sure I'll use https://github.com/sergiusens/snapcraft-preload/commit/d654bbe8a5add5a4bea14d96342b656990c5c818 since that's the most recent commit and also the one that adds `SNAPCRAFT_PRELOAD_REDIRECT_ONLY_SHM`, but also agreed there's not a lot of documentation for snapcraft-preload.

Reply:
done